flow/Platform.actor.cpp

```diff
@@ -0,0 +1,209 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifdef _WIN32
+// This has to come as the first include on Win32 for rand_s() to be found
+#define _CRT_RAND_S
+#include <stdlib.h>
+#include <math.h> // For _set_FMA3_enable workaround in platformInit
+#endif
+
+#include "flow/Platform.h"
+#include "flow/Platform.actor.h"
+#include "flow/Arena.h"
+#include "flow/Trace.h"
+#include "flow/Error.h"
+#include "flow/Knobs.h"
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <cstring>
+#include <algorithm>
+
+#include "flow/FaultInjection.h"
+
+#ifdef _WIN32
+#define NOMINMAX
+#include <windows.h>
+#include <winioctl.h>
+#include <io.h>
+#include <psapi.h>
+#include <stdio.h>
+#include <conio.h>
+#include <direct.h>
+#include <pdh.h>
+#include <pdhmsg.h>
+#pragma comment(lib, "pdh.lib")
+
+// for SHGetFolderPath
+#include <ShlObj.h>
+#pragma comment(lib, "Shell32.lib")
+
+//#define CANONICAL_PATH_SEPARATOR '\\'
+//#define PATH_MAX MAX_PATH
+#endif
+
+#ifdef __unixish__
+#include <dirent.h>
+#include "flow/stacktrace.h"
+#endif
+
+#include "flow/actorcompiler.h"  // This must be the last #include.
+
+
+using std::vector;
+
+Future<vector<std::string>> listFilesAsync( std::string const& directory, std::string const& extension ) {
+	return findFiles( directory, extension, false /* directoryOnly */, true );
+}
+
+Future<vector<std::string>> listDirectoriesAsync( std::string const& directory ) {
+	return findFiles( directory, "", true /* directoryOnly */, true );
+}
+
+ACTOR Future<Void> findFilesRecursivelyAsync(std::string path, vector<std::string> *out) {
+	// Add files to output, prefixing path
+	state vector<std::string> files = wait(listFilesAsync(path));
+	for(auto const &f : files)
+		out->push_back(joinPath(path, f));
+
+	// Recurse for directories
+	state vector<std::string> directories = wait(listDirectoriesAsync(path));
+	for(auto const &dir : directories) {
+		if(dir != "." && dir != "..")
+			wait(findFilesRecursivelyAsync(joinPath(path, dir), out));
+	}
+    return Void();
+}
+
+
+#ifdef _WIN32
+
+
+bool acceptFile( FILE_ATTRIBUTE_DATA fileAttributes, std::string const& name, std::string const& extension ) {
+	return !(fileAttributes & FILE_ATTRIBUTE_DIRECTORY) && StringRef(name).endsWith(extension);
+}
+
+bool acceptDirectory( FILE_ATTRIBUTE_DATA fileAttributes, std::string const& name, std::string const& extension ) {
+	return (fileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0;
+}
+
+ACTOR Future<vector<std::string>> findFiles( std::string directory, std::string extension,
+                                                  bool directoryOnly, bool async) {
+	INJECT_FAULT( platform_error, "findFiles" );
+	state vector<std::string> result;
+    state int64_t tsc_begin = __rdtsc();
```

Comment:
There is a mixture of tabs and spaces throughout this function

Reply:
Will clean this up